### PR TITLE
Add pagination component

### LINKED ISSUE #1391: Build pagination component
## Design spec

- Design and mark down file can be found [here](https://github.com/ubuntudesign/vanilla-design/tree/master/Pagination)

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Add pagination component

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/examples/patterns/pagination/
- Cross reference with [the design spec](https://github.com/ubuntudesign/vanilla-design/tree/master/Pagination)

## Details

Fixes #1391

### CODE REVIEW

**Comment by kwm14** [comment]
## Design review

Only one small change to the font size on large screen which is currently 1.125rem (18px)

### Large screens
- [ ] Pagination title font size to: 1.25rem (20px)

**Comment by kwm14** [comment]
Thanks @barrymcgee looking good 👍

File: examples/patterns/pagination.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+layout: default
+title: Pagination
+category: _patterns
+---
+
+<footer class="p-pagination">
+  <a class="p-pagination__link" href="#previous">
+    <span class="p-pagination__arrow">
+      <img src="https://assets.ubuntu.com/v1/e8d2e45f-chevron-left.svg" alt="">
```
Comment (anthonydillon) [inline examples/patterns/pagination.html]: We need to make all patterns work offline. To be consistent with other patterns with permanent image styling can you add these to a pseudo selector of the links as base64 backgrounds, please.

File: examples/patterns/pagination.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    <div class="p-pagination__text">
+      <span class="p-pagination__label">Previous</span>
+      <span class="p-pagination__title">Lorem ipsum dolor sit amet</span>
+    </div>
+  </a>
+  <a class="p-pagination__link" href="#next">
+    <span class="p-pagination__label">Next</span>
+    <span class="p-pagination__title">Consectetur adipisicing elit</span>
+    <span class="p-pagination__arrow">
+      <img src="https://assets.ubuntu.com/v1/9716bbab-chevron.svg" alt="">
```
Comment (anthonydillon) [inline examples/patterns/pagination.html]: Same here

File: scss/_patterns_pagination.scss [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      &:first-child {
+        padding-left: $sp-x-large;
+
+        @media (max-width: $breakpoint-x-small) {
+          display: block;
+          max-width: $sp-x-small;
+        }
+      }
+
+      &:first-child .p-pagination__arrow {
```
Comment (anthonydillon) [inline scss/_patterns_pagination.scss]: Can you nest this styling in the `&:first-child` above.

File: scss/_settings_breakpoints.scss [hunk 4]
```diff
@@ -1,4 +1,5 @@
 // Global breakpoints
+$breakpoint-x-small: 460px !default;
```
Comment (anthonydillon) [inline scss/_settings_breakpoints.scss]: Can you document this new variable, please.

File: scss/_patterns_pagination.scss (hunk 3 above)
Comment (barrymcgee) [reply scss/_patterns_pagination.scss]: That'll break the linter... we only allow 3 levels deep.

Comment (anthonydillon) [reply scss/_patterns_pagination.scss]: Line 32 is nested.

File: scss/_patterns_pagination.scss [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        padding-left: $sp-x-large;
+
+        @media (max-width: $breakpoint-x-small) {
+          display: block;
+          max-width: $sp-x-small;
+        }
+      }
+
+      .p-pagination__arrow {
+        background-position: 0 0;
```
Comment (anthonydillon) [inline scss/_patterns_pagination.scss]: These styles could move down to line `70`. To keep all arrow styles together.

File: examples/patterns/pagination.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
+---
+layout: default
+title: Pagination
+category: _patterns
+---
+
+<footer class="p-pagination">
+  <a class="p-pagination__link" href="#previous">
+    <span class="p-pagination__arrow"></span>
```
Comment (anthonydillon) [inline examples/patterns/pagination.html]: What's the benefit of having an extra `span` over using pseudo selectors?

File: examples/patterns/pagination.html [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+<footer class="p-pagination">
+  <a class="p-pagination__link" href="#previous">
+    <div class="p-pagination__text">
+      <span class="p-pagination__label">Previous</span>
+      <span class="p-pagination__title">Lorem ipsum dolor sit amet</span>
+    </div>
+  </a>
+  <a class="p-pagination__link" href="#next">
+    <span class="p-pagination__label">Next</span>
```
Comment (anthonydillon) [inline examples/patterns/pagination.html]: Is this one missing: `<div class="p-pagination__text">`?

File: scss/_patterns_pagination.scss [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
...9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHZpZXdCb3g9IjAgMCAzIDEwIj48IS0tIEdlbmVyYXRvcjogU2tldGNoIDQwLjIgKDMzODI2KSAtIGh0dHA6Ly93d3cuYm9oZW1pYW5jb2RpbmcuY29tL3NrZXRjaCAtLT4KIDx0aXRsZT5Hcm91cDwvdGl0bGU+CiA8ZGVzYz5DcmVhdGVkIHdpdGggU2tldGNoLjwvZGVzYz4KIDxnIGlkPSJQYWdlLTEiIGZpbGwtcnVsZT0iZXZlbm9kZCIgZmlsbD0ibm9uZSI+CiAgPGcgaWQ9Ikdyb3VwIiBmaWxsPSIjMzMzIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxLjUgNSkgc2NhbGUoLTEgMSkgdHJhbnNsYXRlKC0xLjUgLTUpIj4KICAgPGcgaWQ9IlNoYXBlIj4KICAgIDxwYXRoIGQ9Im0wIDEwdi0xLjc1bDAuMDYyLTAuMDljMC4xODYtMC4yNjYgMC4zNjgtMC41NCAwLjU0NS0wLjgyMyAwLjIyNS0wLjM2IDAuNDM2LTAuNzI3IDAuNjMzLTEuMTA0IDAuMTktMC4zNjQgMC40MjMtMC44NzQgMC41NzgtMS4yMzMtMC4xNTUtMC4zNi0wLjM4Ny0wLjg3LTAuNTc4LTEuMjM0LTAuMTk3LTAuMzc2LTAuNDA4LTAuNzQ0LTAuNjMzLTEuMTA0LTAuMTc3LTAuMjg0LTAuMzYtMC41NTYtMC41NDUtMC44MjNsLTAuMDYyLTAuMDg5di0xLjc1bDAuNTg0IDAuNzdjMC4zMjQgMC40NSAwLjYzNyAwLjkwNiAwLjkyNyAxLjM3NCAwLjMyNSAwLjUyMyAwLjYxNiAxLjAzIDAuODczIDEuNTI0IDAuMjM4IDAuNDU2IDAuNDMyIDAuODggMC41NiAxLjE5NmwwLjA1NiAwLjEzNi0wLjA1NiAwLjEzNmMtMC4xMzMgMC4zMjQtMC4zMjMgMC43NC0wLjU2IDEuMTk1LTAuMjU4IDAuNDk1LTAuNTUgMS4wMDMtMC44NzMgMS41MjUtMC4yOSAwLjQ2OC0wLjYwMiAwLjkyNi0wLjkyNiAxLjM3NmwtMC41ODUgMC43Njh6Ii8+CiAgIDwvZz4KICA8L2c+CiA8L2c+Cjwvc3ZnPgo=');
```
Comment (anthonydillon) [inline scss/_patterns_pagination.scss]: To reduce size can we reuse this icon and use rotation to point in the other direction?

File: examples/patterns/pagination.html (hunk 7 above)
Comment (barrymcgee) [reply examples/patterns/pagination.html]: No, that wrap is required on the first link becuase the text is `display: none` on small screens.

File: scss/_patterns_pagination.scss (hunk 8 above)
Comment (barrymcgee) [reply scss/_patterns_pagination.scss]: Ooooh - good call 👌🏻

File: examples/patterns/pagination.html (hunk 7 above)
Comment (anthonydillon) [reply examples/patterns/pagination.html]: Don't you think it would be nice to have them the same to symmetry?

**Pull request merged**